packages/tabs/src/Tabs.ts

```diff
@@ -245,16 +245,55 @@ export class Tabs extends SizedMixin(Focusable, { noDefaultSize: true }) {
         return complete;
     }
 
-    public async scrollToSelection(): Promise<void> {
+    private getNecessaryAutoScroll(index: number): number {
+        const selectedTab = this.tabs[index];
+        const selectionEnd = selectedTab.offsetLeft + selectedTab.offsetWidth;
+        const viewportEnd = this.tabList.scrollLeft + this.tabList.offsetWidth;
+        const selectionStart = selectedTab.offsetLeft;
+        const viewportStart = this.tabList.scrollLeft;
+
+        // If the tab is already visible, the necessary scroll factor is 0.
+        let scrollTarget = 0;
+
+        if (selectionEnd > viewportEnd) {
+            // Selection is on the right side, not visible.
+            const nextTab = this.tabs[index + (this.dir === 'rtl' ? -1 : 1)];
+            scrollTarget = nextTab
+                ? nextTab.offsetLeft - this.tabList.offsetWidth
+                : viewportEnd;
```

Comment:
We can modularize this functions for better testing approach!

Reply:
Outside of this class you mean?

Reply:
Yes!

Reply:
I added some modifications here, found a more straight forward approach to account for tabs margins, let me know if you think it is better.

Reply:
Would you suggest where woult be a good place for it?

Reply:
You can create a helper method here also! We will think where can we best put this later on!